packages/react-components/src/components/ActionBar/ActionBar.tsx

```diff
@@ -24,128 +24,83 @@ export const ActionBar: React.FC<IActionBarProps> = ({
   vertical,
   menuFooter,
 }) => {
-  const [menuItemsKeys, setMenuItemsKeys] = React.useState<string[]>([]);
-  const [menuPosition, setMenuPosition] = React.useState<number>(0);
+  const [visibleItemsCount, setVisibleItemsCount] = React.useState<number>(
+    options.length
+  );
+  const [menuOptions, setMenuOptions] = React.useState<IActionBarOption[]>([]);
   const [isMenuOpen, setIsMenuOpen] = React.useState<boolean>(false);
   const isScrollType = type === 'scroll';
   const mergedClassNames = cx(
     styles[baseClass],
     className,
     vertical && styles[`${baseClass}--vertical`]
   );
-  const observerOptions = {
-    root: document.querySelector(`${id}`),
-    threshold: 1,
-  };
-  const shouldDisplayMenu = !isScrollType && menuItemsKeys.length !== 0;
+  const singleElementSize = 44;
```

Comment:
Shouldn't it be available as a prop? If not I'd suggest to move it outside the component's function.

Reply:
Currently not, but in the future I plan to improve that component and give a possibility to set the buttons size.